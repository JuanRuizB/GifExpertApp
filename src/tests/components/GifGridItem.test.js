import { GifGridItem } from "../../components/GifGridItem"
import { shallow } from "enzyme"


describe('Pruebas en <GifGridItem', () => {

    const title = "Titulo de la prueba";
    const url = "https://media3.giphy.com/media/l0Iyoy5VglR6QKJiw/giphy.gif?cid=22b370e9r9q8j3rzb3l8rxm1yg82vbp9y17h27ru31n6sed8&rid=giphy.gif";
    const wrapper = shallow(<GifGridItem title={ title } url={ url }/>);
    
    test('Debe de mostrar el componente correctamente', () => {
        
        expect( wrapper ).toMatchSnapshot();

    });
    
    test('Debe de tener un pÃ¡rrafo con el title', () => {
        
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );

    });

    test('Debe de tener la imagen igual al url y alt de los props', () => {
        
        const img = wrapper.find('img');
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );

    });

    test('debe de tener animate__fadeIn', () => {
        
        const div = wrapper.find('div');
        const className = div.prop('className');

        expect( className.includes("animate__fadeIn") ).toBe( true );

    })
    
    
    


})
