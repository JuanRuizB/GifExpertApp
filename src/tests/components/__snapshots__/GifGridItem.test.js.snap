// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem Debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo de la prueba"
    src="https://media3.giphy.com/media/l0Iyoy5VglR6QKJiw/giphy.gif?cid=22b370e9r9q8j3rzb3l8rxm1yg82vbp9y17h27ru31n6sed8&rid=giphy.gif"
  />
  <p>
    Titulo de la prueba
  </p>
</div>
`;
